cmd/soroban-rpc/docker/k8s_deployment.yml

```diff
@@ -0,0 +1,128 @@
+# running captive core with on-disk mode limits RAM to around 2G usage, but 
+# requires some dedicated disk storage space that has at least 3k IOPS for read/write.
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: soroban-rpc-futurenet-core-storage
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 500Gi
+  storageClassName: default
+  volumeMode: Filesystem  
+---   
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    fluxcd.io/ignore: "true"
+  labels:
+    app: soroban-rpc-futurenet
+  name: soroban-rpc-futurenet
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: soroban-rpc-futurenet
+  template:
+    metadata:
+      annotations:
+        fluxcd.io/ignore: "true"
+      creationTimestamp: null
+      labels:
+        app: soroban-rpc-futurenet
+    spec:
+      initContainers:
+      - name: clean-volume
+        image: busybox:1.28
+        command: ["rm", "-rf", "/opt/stellar/soroban-rpc/captive-core"]
+        volumeMounts:
+        - mountPath: /opt/stellar/soroban-rpc
+          name: captive-core-data
+      containers:    
+      - name: rpc-server
+        image: stellar/quickstart:soroban-dev@sha256:81c23da078c90d0ba220f8fc93414d0ea44608adc616988930529c58df278739
+        args: ["--futurenet", "--enable-soroban-rpc", "--logs"]
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 8000
+          name: web
+          protocol: TCP
+        readinessProbe:
+          exec:
+              command:
+                - /bin/sh
+                - -c
+                - | 
+                  curl --location --request POST 'http://localhost:8000/soroban/rpc' \
+                    --header 'Content-Type: application/json' \
+                    --data-raw '{
+                      "jsonrpc": "2.0",
+                      "id": 10235,
+                      "method": "getHealth"
+                    }' | jq --exit-status 'has("jsonrpc")' 
```

Comment:
Nice. Is this recommended way to check health of the service? If so I'd like to incorporate it in the helm chart too

Reply:
for readiness probe concerns, this should be adequate, it validates json-rpc service is responsive, it doesn't check further functional readiness, i.e. the response of `getHealth` , which could be `result:healthy` or `error` depending on the underlying ingestion state which rpc has ongoing with it's captive core, but in that case we don't want to allow k8s to restart the service due to probe failures as the error response means ingestion is in-process of catch up.